Cobertura Check additionally at code testing.

levilime opened the pull request:
[![Build Status](https://travis-ci.org/levilime/tygron.svg?branch=CI-speed-improvement)](https://travis-ci.org/levilime/tygron)

The current pom configuration doesn't have cobertura as goal when testing, so testing should be invoked like this to also run cobertura.

levilime committed:
Cobertura Check additionally at code testing.

The current pom configuration doesn't have cobertura as goal when testing, so testing should be invoked like this.

MaxGroenenboom commented:
I think `mvn cobertura:check` succeeds when tests fail but the coverage is still high enough... Also this PR should fix #101

levilime committed:
Readded both test run and cobertura check seperately.

Otherwise the build would not fail if coverage is too low.
Also added caching for maven, which should help the speed of building

MaxGroenenboom commented:
This should be merged asap or our coverage might start decreasing. I approve.

spiderbiggen commented:
Approved

-RAMpestampers

jannelie commented:
Approved -CrashTestDummies

MHMCD commented:
Approved
- CodeFox ![CodeFox](https://avatars1.githubusercontent.com/u/18596181?v=3&s=32)
